Cubical/Data/FinSet.agda

```diff
@@ -0,0 +1,110 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+
+module Cubical.Data.FinSet where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Logic
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.Equiv
+open import Cubical.HITs.PropositionalTruncation
+open import Cubical.Data.Unit
+open import Cubical.Data.Nat
+open import Cubical.Data.Fin
+open import Cubical.Data.Sigma
+
+private
+  variable
+    ℓ : Level
+    A : Type ℓ
+
+isFinSet : Type ℓ → Type ℓ
+isFinSet A = ∃[ n ∈ ℕ ] A ≃ Fin n
+
+isProp-isFinSet : isProp (isFinSet A)
+isProp-isFinSet = propTruncIsProp
+
+FinSet : Type (ℓ-suc ℓ)
+FinSet = TypeWithStr _ isFinSet
+
+isFinSetFin : ∀ {n} → isFinSet (Fin n)
+isFinSetFin = ∣ _ , pathToEquiv refl ∣
+
+isFinSetUnit : isFinSet Unit
+isFinSetUnit = ∣ 1 , Unit≃Fin1 ∣
+  where
+    Unit≃Fin1 : Unit ≃ Fin 1
```

Comment:
Maybe this equivalence should be upstreamed? I can imagine it being useful elsewhere

Reply:
`FinSet` depends on `Fin` and `Fin` depends on `Unit`. So if I import `FinSet` from `Unit` I get a cycle.

Reply:
Oh, I misunderstood. You only meant the equivalence. But I have tried to only move the equivalence and I still get a cycle, although a less obvious one: `Unit -> Fin -> Nat.Order -> ... -> Unit`.

Reply:
Right, I just meant the equivalence. Can't you put it in Fin?

Reply:
That works

Reply:
Perfect! That seems like a reasonable place to put it